### DISCUSSION TITLE
when regiter user always return error id is not defined

### DISCUSSION THREAD

**Discussion by wahyono17** [adonisjs/core#3416] [Help] [answered]
when i register user is always return error id is not defined, but actually register is succeed, 
when i call await auth.use("api").login(email,password) is always return is is not defined
when i look inside the function, the function is in node_modules\\@adonisjs\\auth\\build\\src\\Guards\\Base\\index.js
but what is id i have to provide ?

here is the complete return response 
Exception: Cannot login user. Value of \"id\" is not defined\n    at OATGuard.getUserForLogin (C:\\Users\\HP 840\\Documents\\adonisProject\\vue-adonis\\server-ts\\node_modules\\@adonisjs\\auth\\build\\src\\Guards\\Base\\index.js:104:19)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at OATGuard.login (C:\\Users\\HP 840\\Documents\\adonisProject\\vue-adonis\\server-ts\\node_modules\\@adonisjs\\auth\\build\\src\\Guards\\Oat\\index.js:235:30)\n    at AuthController.register (C:\\Users\\HP 840\\Documents\\adonisProject\\vue-adonis\\server-ts\\app\\Controllers\\Http\\AuthController.ts:35:9)\n    at Object.PreCompiler.runRouteHandler [as fn] (C:\\Users\\HP 840\\Documents\\adonisProject\\vue-adonis\\server-ts\\node_modules\\@adonisjs\\http-server\\build\\src\\Server\\PreCompiler\\index.js:47:31)\n    at Server.handleRequest (C:\\Users\\HP 840\\Documents\\adonisProject\\vue-adonis\\server-ts\\node_modules\\@adonisjs\\http-server\\build\\src\\Server\\index.js:108:13)"

**Comment by RomainLanz** [comment, marked as answer]
Hey @wahyono17! 👋🏻 

Could you please provide some code?
It is quite hard to help you without any more information.

**Reply by wahyono17** [reply]
public async register({request,auth,response}){
        const { email, password, confirm_password } = request.all();

        await User.create({
            email,
            password,
        });

        return await auth.use("api").login(email,password);
    }

**Reply by thetutlage** [reply]
The `login` method does not accept the user email and password. Please read the docs carefully
